pkg/handlers/default.go

```diff
@@ -49,3 +56,68 @@ func KillContainer(pid int) registry.HandlerFunc {
 		return registry.HandlerResultErrno(unix.EPERM)
 	}
 }
+
+func FreezeContainer(pid int) registry.HandlerFunc {
+	re := regexp.MustCompile(`^(\d*):(freezer)?:(/..*)$`)
+	return func(fd libseccomp.ScmpFd, req *libseccomp.ScmpNotifReq) (result registry.HandlerResult) {
+		cgroupPath := ""
+		if cgroupFile, err := os.Open(filepath.Join("/proc", fmt.Sprintf("%d", pid), "cgroup")); err == nil {
+			defer cgroupFile.Close()
+			reader := bufio.NewReader(cgroupFile)
+			for {
+				line, err := reader.ReadString('\n')
+				if err != nil {
+					break
+				}
+				line = strings.TrimSuffix(line, "\n")
+
+				r := re.FindStringSubmatch(line)
+				if len(r) < 4 {
+					continue
+				}
+
+				cgroupPath = r[3]
+				break
+			}
+		} else {
+			log.WithFields(log.Fields{
+				"pid": pid,
+				"err": err,
+			}).Error("cannot parse cgroup")
+			return registry.HandlerResultErrno(unix.EPERM)
+		}
+		if cgroupPath == "" {
+			log.WithFields(log.Fields{
+				"pid": pid,
+			}).Error("cgroup path not found")
+			return registry.HandlerResultErrno(unix.EPERM)
+		}
+
+		cgroup := &configs.Cgroup{
+			Path:      cgroupPath,
+			Resources: &configs.Resources{},
+		}
+
+		m, err := manager.New(cgroup)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"pid":        pid,
+				"cgroupPath": cgroupPath,
+				"err":        err,
+			}).Error("cannot create new cgroup manager")
+			return registry.HandlerResultErrno(unix.EPERM)
+		}
+
+		err = m.Freeze(configs.Frozen)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"pid":        pid,
+				"cgroupPath": cgroupPath,
+				"err":        err,
+			}).Error("cannot freeze cgroup")
+			return registry.HandlerResultErrno(unix.EPERM)
+		}
+
+		return registry.HandlerResultErrno(unix.EPERM)
```

Comment:
On success we return EPERM? Why? How are the callers using this return value?

Reply:
This PR never thaw the container, so it stays frozen, so this return value is never actually used. But if the alert is sent to an external system, and the container is thaw by that external system, I guess we still don't want the syscall to be allowed.

Reply:
Ohh, I see how the return is used. It is used for the response for the syscall. Looks fine then, thanks!